### ISSUE TITLE
[Feature]: Feature: Zone-level alarm_control_panel entities for spaces with Zone Mode enabled

### ISSUE THREAD

**Issue by Cingar01** [bvis/aegis-hass#86]
### Description

When Ajax Zone Mode is enabled on a space, Aegis currently creates a single 
`alarm_control_panel` entity for the entire space. This makes it impossible 
to trigger automations based on which specific zone was armed/disarmed.

### Use case

I have a space with Zone Mode enabled and two zones:
- **Zone 1** (ID 01, 32 devices) — my primary residence
- **Zone 2** (ID 02, 7 devices) — tenant apartment

The tenant manages their own zone independently. I cannot arm the entire 
space simultaneously since the tenant may be home.

**Current problem:** When I arm only Zone 1, I cannot trigger automations 
specifically for that event because:
1. The single `alarm_control_panel` entity does not reflect zone-level state
2. The `event.security_event` entity receives FCM events but has no 
   `zone_name` or `zone_id` attribute to filter on

### Related Ajax devices

Superior Hub Hybrid 4G (Zone Mode)

### Current Setup
- Hub: Superior Hub Hybrid 4G
- Zone Mode: Enabled
- Zones: Zone 1 (ID 01, 32 devices), Zone 2 (ID 02, 7 devices)
- Aegis version: v1.2.3
- App label: Ajax (standard, com.ajaxsystems)
- FCM: Configured and credentials verified

### Why This Matters
Without zone-level separation, common automations become impossible:
- Turn off power outlets only when Zone 1 is armed (not tenant's Zone 2)
- Send notifications specific to which zone triggered an alarm
- Automate based on home/away status per zone independently

Thank you for the excellent work on this integration!

**Comment by bvis** [comment]
@Cingar01 I've put up a draft PR for the per-zone alarm panel work: #87 — and just shipped it as **v1.2.4-beta.1** so you can install it via HACS instead of a branch.

Looking at the protos, "Zone Mode" in the newer firmware UI is the same underlying feature as the older "Group Mode" — the gRPC payload is `GroupSecurity` in both cases — so the implementation covers both. With your setup (Zone 1 / 32 devices and Zone 2 / 7 devices) you should see two `alarm_control_panel` entities after installing the beta, one per zone.

Could you install the beta and validate the first half of your issue (the per-zone panels)? Steps:

1. HACS → ⋮ → **Settings** → enable **Show beta versions**.
2. HACS → Integrations → Aegis for Ajax → ⋮ → **Redownload** → in the version dropdown choose **1.2.4-beta.1**, then restart Home Assistant.
3. After restart, check that you see one `alarm_control_panel` entity per zone (Villa and Apartment), with names matching your Ajax app.
4. Each entity exposes `group_id`, `group_name`, `space_id`, `hub_id` as state attributes — those are what you'd filter on in your "only when MY zone arms" automations.
5. Test arming/disarming each zone from HA and from the Ajax app, and confirm both directions reflect correctly. Note: remote arming from the Ajax app may take up to one hour to reflect in HA in this initial slice (intentional, see release notes) — local HA → Ajax should be immediate.

The second half of your issue — `event.security_event` carrying `zone_id`/`zone_name` — is **not** in this beta. To implement it properly I need the FCM push payload that arrives when a zone arm/disarm fires. Once you have FCM configured (per the README clarification we merged in #85), if you could grab a debug-level log of `custom_components.aegis_ajax` while arming Zone 1 from the Ajax app, it would let me verify whether the zone identifier comes through in the push and where, and ship that as a follow-up.

Release page (full notes + known limitations): https://github.com/bvis/aegis-hass/releases/tag/v1.2.4-beta.1

**Comment by Cingar01** [comment]
Hi @bvis, thanks for the quick work on #87!
I installed the feat/group-mode-alarm-panels branch via HACS and restarted.
What works:

  *   Two new zone-level alarm_control_panel entities appeared — one for Villa (32 devices) and one for Apartment (7 devices). Names match the Ajax app. However, the original whole-house panel seems to have broken — ideally there should be three panels: whole house, Villa, and Apartment. That would give full flexibility to arm everything at once or each zone independently.

Issues found:

  1.  All three entities show unavailable — both zone entities and the original whole-house entity are stuck in unavailable state. The original entity also has restored: true, suggesting it didn't initialize properly after restart.
  2.  Missing state attributes — the zone entities only expose friendly_name and supported_features: 2. The group_id, group_name, space_id, and hub_id attributes mentioned in the PR are not present.
  3.  Arming fails with protobuf error — attempting to arm the Villa zone (Away) returns:

Failed to perform the action alarm_control_panel/alarm_arm_away.
module 'systems.ajax.api.mobile.v2.space.security.group.arm_group_request_pb2'
has no attribute 'ArmGroupRequest'

  1.  alarm_arm_night not supported — supported_features: 2 means only arm_away is exposed. Any plans to add night mode / perimeter protection (alarm_arm_night)? We use this daily — perimeter sensors armed while indoor motion detectors stay disarmed.

Happy to test further. Let me know if you need anything from my end.

Best Regards

________________________________

**Comment by bvis** [comment]
@Cingar01 thanks for the thorough report — exactly what was needed. Three of the four points were real bugs and they're now fixed in **v1.2.4-beta.2**:

1. **Arm error** (`module ... has no attribute 'ArmGroupRequest'`) — the proto class was actually `ArmSpaceGroupRequest`. My unit tests for `arm_group` mocked the proto module wholesale (`MagicMock()`), so the typo silently passed. Added a real-proto regression suite that would have caught this in `beta.1`. Fixed in #88.
2. **Entities stuck `unavailable` + no state attributes** — coordinator's `list_spaces()` poll was wiping the cached groups every cycle, so per-group panels saw `_group is None` and went unavailable. The cached groups + `group_mode_enabled` are now preserved across polls in the same merge step that already preserves monitoring companies. Fixed in #88.
3. **The whole-house panel** — you're right, it shouldn't disappear in group mode. `beta.1` replaced it with per-group panels only, leaving the v1.2.3 entity orphaned in your registry as `restored: true`. The integration now always creates the whole-house panel **and** adds per-group panels alongside when Group Mode is on. So you should see **1 + 2 = 3 panels** working: whole-house (with night mode) + Villa + Apartment. Fixed in #88.

For the fourth point (**night mode on per-group panels**) — this is a protocol limitation, not an oversight. Ajax's gRPC service exposes `armToNightMode` (whole space) and `armGroup` (no night flag) but no `armGroupToNightMode`. So per-group night mode isn't representable. Now that the whole-house panel is back, you get night mode there as before — fires `armToNightMode` for the whole space. If your daily-use pattern was actually night mode on Villa only while Apartment stays disarmed, that's not something I can implement against the current Ajax API; it would need a hub-side change from Ajax themselves.

Steps to test:

1. HACS → ⋮ → Settings → ensure **Show beta versions** still on.
2. HACS → Aegis for Ajax → ⋮ → **Redownload** → choose **1.2.4-beta.2** → restart HA.
3. After restart, check that you see three `alarm_control_panel` entities:
   - The whole-house one (whatever it was named in v1.2.3 — should no longer be `restored: true`).
   - `alarm_control_panel.villa`
   - `alarm_control_panel.apartment`
4. Try arming Villa from HA → expect success this time.
5. Verify state attributes on the zone panels include `group_id`, `group_name`, `space_id`, `hub_id`.
6. Arm/disarm each zone from the Ajax app and confirm HA reflects within reasonable time (still up to one hour for full sync since the long-lived stream subscription is a follow-up).

Release page: https://github.com/bvis/aegis-hass/releases/tag/v1.2.4-beta.2

**Comment by bvis** [comment]
Closing this one optimistically — zone-level `alarm_control_panel` entities for spaces in Ajax Zone Mode have been live since **v1.2.4-beta.1** (PR #87, sharing the same `GroupSecurity` proto with Group Mode), with three end-to-end fixes on top in **beta.2** (PR #88) addressing the issues you flagged: arming a group failed with the proto-class typo, panels were flapping to `unavailable` between snapshot polls, and the whole-house panel was disappearing when Zone Mode was enabled. The latest in the channel is **v1.2.4-beta.4**.

Hi @Cingar01 — thanks again for the very thorough hardware testing on the Hub Hybrid 4G; without it `beta.1` would have shipped with all three of those bugs unnoticed. If anything still feels off on the latest beta — group state not updating after arming from the Ajax app (the snapshot path takes up to 1h, that's known), entity attributes missing what your automations need, anything else — please **reopen this issue** with debug logs and we'll keep iterating.

The follow-up of enriching the FCM `event.aegis_security_event` entity with `zone_id` / `zone_name` is intentionally still out — that one needs the FCM payload that arrives when arming a single zone, which I can't reproduce locally. If you happen to capture that payload at debug level (`firebase_messaging`-related lines around an arm-by-zone), drop it on a fresh issue and I'll wire the parser.
